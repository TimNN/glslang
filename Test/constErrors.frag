#version 330

in vec4 inVar;
out vec4 outVar;

const int constInt = 3;

uniform int uniformInt;

void main()
{
    const int a1 = 2;          // okay
    const int a2 = constInt;   // okay
    const int a3 = uniformInt; // error

    vec4 c[constInt];              // okay
    vec4 d[uniformInt];            // error
    vec4 e[constInt + uniformInt]; // error
    vec4 f[uniformInt + constInt]; // error

    vec4 g[sin(3.2)];              // okay
}
