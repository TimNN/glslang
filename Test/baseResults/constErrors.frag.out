Warning, version 330 is not yet complete; some version-specific features are present, but many are missing.
ERROR: 0:14: '=' :  assigning non-constant to 'const int'
ERROR: 0:17: '' : constant expression required 
ERROR: 0:17: '' : array size must be a constant integer expression 
ERROR: 0:18: '' : constant expression required 
ERROR: 0:18: '' : array size must be a constant integer expression 
ERROR: 0:19: '' : constant expression required 
ERROR: 0:19: '' : array size must be a constant integer expression 
ERROR: 0:27: '=' :  assigning non-constant to 'const structure'
ERROR: 0:33: '=' :  assigning non-constant to 'const structure'
ERROR: 9 compilation errors.  No code generated.

ERROR: node is still EOpNull!
0:10  Function Definition: main( (void)
0:10    Function Parameters: 
0:?   Linker Objects
0:?     'inVar' (smooth in 4-component vector of float)
0:?     'outVar' (out 4-component vector of float)
0:?     'constInt' (const int)
0:?       3 (const int)
0:?     'uniformInt' (uniform int)
0:?     's' (structure)
0:?     's2' (structure)

